### ISSUE TITLE
bafy addresses ...

### ISSUE THREAD

**Issue by michel47** [ipfs/go-ipfs#7410]
#### Version information:

```
go-ipfs version: 0.5.1
Repo version: 9
System version: amd64/linux
Golang version: go1.13.10
```

#### Description:
After switching to ipfs 0.5 and fixing a few API changes, I still have problem with some of multihashes due to the fact that old path [http://localhost:8080/ipfs/:mhash58](http://localhost:8080/ipfs/zPx88Lhfqso4mGnfRGeqV8o2axRxFLGNfzwMaDxDNfDpu) get systematically redirected to  [http://:bafy32.ipfs.localhost:8080/](http://bafkqahljomqhi2djomqho33snnuw4zzao4xsaskqizjsambogu7qu.ipfs.localhost:8080/)

and this breaks w/ https://gateway.ipfs.io/ipfs/mAXAAMxItChgBcBEU7+JFmSW5JqeOINP+HJJInW4UG7QSDWJnLW1lYnViLndlYnAYtMYBCgIIAQ/bg-mebub.webp 

The error I get is
```
bafyaamysfufbqalqceko7ysftes3sjvhryqnh7q4sjej23qudo2bedlcm4ww2zlcovrc453fmjybrnggaefaecab.ipfs.localhost’s server IP address could not be found.
```

would it be because of the length of the "bafy" multihash ?
because https://ipfs.io/ipfs/z8bvj226WBEd2XS2mnGgjd9FrrAb1mWfd is working fine
and so does : http://bafybcfhp4jczsjnze2ty4igt7yojese5nykbxna.ipfs.localhost:8080/

if I access [/ipfs/mAXAAMxItChgBcBEU7+JFmSW5JqeOINP+HJJInW4UG7QSDWJnLW1lYnViLndlYnAYtMYBCgIIAQ/bg-mebub.webp](https://cloudflare-ipfs.com/ipfs/mAXAAMxItChgBcBEU7+JFmSW5JqeOINP+HJJInW4UG7QSDWJnLW1lYnViLndlYnAYtMYBCgIIAQ/bg-mebub.webp)
the via an old 0.4 server (like cloudflare's one) : it does works (probably because it doesn't switch to the "subdomained" address)

To Summarize 
```sh
# this works :
xdg-open http://localhost:8080/ipfs/$(echo "is this working w/ IPFS 0.5 ? YES" | ipfs add -Q --hash identity)
# this doesn't :
xdg-open http://localhost:8080/ipfs/$(echo "is this working with IPFS 0.5 ? NO" | ipfs add -Q --hash identity)
```

Can we have a conditional redirect if bafy hash > 63char then don't redirect ?

**Comment by Stebalien** [comment]
Thanks for the report. This looks like a duplicate of #7318.

This will be mostly fixed by https://github.com/ipfs/go-ipfs/pull/7358 but long CIDs will still cause problems on gateways using HTTPS because the solution involves multiple subdomains.

Locally, you can use `127.0.0.1` instead of `localhost` to disable this or configure your node to disable this feature:

```
{
   ...
  "Gateway": {
    "PublicGateways": {
      "localhost": {
        "Paths": ["/ipfs", "/ipns"],
        "UseSubdomains": false
      }
    }
  }
}
```

**Comment by ribasushi** [comment]
@michel47 out of curiosity - what software / series of steps created these CIDs? I am specifically wondering what is out there in the wild that makes 408-byte long identity CIDs like your example of  https://cid.ipfs.io/#bafyaamysfufbqalqceko7ysftes3sjvhryqnh7q4sjej23qudo2bedlcm4ww2zlcovrc453fmjybrnggaefaecab

**Comment by Stebalien** [comment]
He used `ipfs add --inline --inline-length=...`.

**Comment by michel47** [comment]
Thanks @stebalien, I was not aware of the --inline option, and yes it is pratically what I was doing as I am using a program call wrapid which create a "dag-object" wrapper to tag an immutable hash with a name (same as ipfs add -w --hash SHA1) but with a identity hashing such that the wrapper is in the URL.

sha1 is used as it is the smallest hash allowed thus minimizing the size of the "inline" object ...
so the length of the object is basically length(label)+36 bytes

```
{"data":"CAE=","links":[{"Name":"label678","Size":0,"Cid":{"/":"bafkrcfemdrgiu6s3qog46qnt4r7aifddskrxh6i"}}]}
```

**Comment by Stebalien** [comment]
Whatever you do, please do not use sha1. It's supported for compatibility with git but it's not secure against some forms of attacks and will likely be disabled/restricted in the future.
